### ISSUE TITLE
vmagent: remote-write retries "hammer" vminserts

### ISSUE THREAD

**Issue by perfect-duo** [VictoriaMetrics/VictoriaMetrics#5486]
### Is your question request related to a specific component?

vmagent vminsert

### Describe the question in detail

2023-12-19T07:43:29.157Z	warn	VictoriaMetrics/app/vmagent/remotewrite/client.go:385	couldn't send a block with size 70070 bytes to "1:https://xxxxx/insert/236/prometheus": Post "https://xxxx/insert/236/prometheus": context deadline exceeded (Client.Timeout exceeded while awaiting headers); re-sending the block in 2.000 seconds

There is a problem with vminsert configuration. Therefore, vmagent keeps failing to send data if there are too many vmagent nodes. This will lead to constant retries, which in turn increases the pressure on our SLB entrance and vminsert. So is there a way to control the retry interval of vmagent?

### Troubleshooting docs

- [ ] General - https://docs.victoriametrics.com/Troubleshooting.html
- [X] vmagent - https://docs.victoriametrics.com/vmagent.html#troubleshooting
- [ ] vmalert - https://docs.victoriametrics.com/vmalert.html#troubleshooting

**Comment by dmitryk-dk** [comment]
Hi @perfect-duo ! This values is limited inside the function and growth twice per each error

```
retryDuration := time.Second
	retriesCount := 0

again:
	startTime := time.Now()
	resp, err := c.doRequest(c.remoteWriteURL, block)
	c.requestDuration.UpdateDuration(startTime)
	if err != nil {
		c.errorsCount.Inc()
		retryDuration *= 2
		if retryDuration > time.Minute {
			retryDuration = time.Minute
		}
		logger.Warnf("couldn't send a block with size %d bytes to %q: %s; re-sending the block in %.3f seconds",
			len(block), c.sanitizedURL, err, retryDuration.Seconds())
		t := timerpool.Get(retryDuration)
```
So at that time it can't be set via any flag

**Comment by hagen1778** [comment]
> There is a problem with vminsert configuration. Therefore, vmagent keeps failing to send data if there are too many vmagent nodes. This will lead to constant retries, which in turn increases the pressure on our SLB entrance and vminsert. 

What settings would you like to set to avoid this?

**Comment by jiekun** [comment]
Close due to inactivity.

**Comment by hagen1778** [comment]
Ability to customize retry policy via `-remoteWrite.retryMinInterval` and `-remoteWrite.retryMaxTime` cmd-line flags was added https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6289 and will be available in the next release.

**Comment by hagen1778** [comment]
This feature was included into [v1.103.0](https://docs.victoriametrics.com/changelog/#v11030).
